### PR TITLE
add qwen3-moe operation

### PR DESCRIPTION
**Pull request by shiyuan680**
### What this PR does / why we need it?
origin qwen3_moe loss alltoall operation which result fault resultt, in this pr reuse some optimizations from deepseek.
### Does this PR introduce _any_ user-facing change?
no user-facing change
### How was this patch tested?

e2e & ut

- vLLM version: v0.9.2
- vLLM main: https://github.com/vllm-project/vllm/commit/b639327ad94b3aa16022ebea49f8e525660b736b

### CODE REVIEW

File: vllm_ascend/models/qwen3_moe.py [hunk 1]
```diff
@@ -33,3 +49,85 @@ class CustomQwen3MoeForCausalLM(Qwen3MoeForCausalLM):
         "experts":
         ["experts.0.gate_proj", "experts.0.up_proj", "experts.0.down_proj"],
     }
+
+
+class AscendQwen3MoeSparseMoeBlock(nn.Module):
```
Comment (ApsarasX) [inline vllm_ascend/models/qwen3_moe.py:54]: What is the issue that this PR aims to address?
